tools/playbooks/build-upload-openstack.yaml

```diff
@@ -0,0 +1,265 @@
+            key_name: "kp-bootc-build-{{ random_num }}"
+            security_groups: "sg-bootc-build-{{ random_num }}"
+            auto_ip: true
+            config_drive: true
+            wait: true
+            timeout: 600
+
+        - name: Makre sure instance is running
+          openstack.cloud.server_info:
+            server: "bootc-image-test-rhel-9-4-{{ firmware }}"
+          register: instance_status_result
+          retries: 20
+          until: instance_status_result.servers[0].vm_state == "active" and instance_status_result.servers[0].access_ipv4 != ""
+
+        - set_fact:
+            instance_ip: "{{ instance_status_result.servers[0].addresses.shared_net_9[0].addr }}"
+          when:
+            - instance_status_result is succeeded
+
+        - name: Waits until instance is reachable
+          wait_for:
+            host: "{{ instance_ip }}"
+            port: 22
+            search_regex: OpenSSH
+            delay: 10
+          retries: 30
+          register: result_ssh_check
+          until: result_ssh_check is success
+
+        - name: add os instance ip into host group builder
+          add_host:
+            name: "{{ instance_ip }}"
+            groups: guest
+
+        - name: Ensure cloud-init has finished
+          raw: test -f /var/lib/cloud/instance/boot-finished
+          retries: 30
+          delay: 20
+          register: cloud_init_check
+          changed_when: false
+          until: cloud_init_check is success
+          delegate_to: "{{ instance_ip }}"
+
+        # https://cloudinit.readthedocs.io/en/latest/topics/modules.html#growpart
+        # 10G image will be resized to 40G (disk size)
+        - name: disk growpart feature checking
+          command: df -h
+          become: yes
+          register: result_growpart
+          delegate_to: "{{ instance_ip }}"
+
+        - name: LVM checking
+          fail:
+            msg: "No LVM support"
+          when: "'/dev/mapper/bootc-root' not in result_growpart.stdout"
+
+      rescue:
+        - name: remove tmp image
+          openstack.cloud.image:
+            name: "tmp-bootc-image-rhel-9-4-{{ firmware }}"
+            state: absent
+
+        - name: "building openstack image fail"
+          fail:
+            msg: "building openstack image fail"
+
+      always:
+        - name: remove image test instance
+          openstack.cloud.server:
+            name: "bootc-image-test-rhel-9-4-{{ firmware }}"
+            state: absent
+
+        - name: delete tested tmp image
+          openstack.cloud.image:
+            name: "tmp-bootc-image-rhel-9-4-{{ firmware }}"
+            state: absent
+
+    - name: "delete last openstack image"
+      openstack.cloud.image:
+        name: "bootc-rhel-9-4-lvm-{{ firmware }}"
+        state: absent
+
+    - name: "upload new openstack image"
+      openstack.cloud.image:
+        name: "bootc-rhel-9-4-lvm-{{ firmware }}"
+        filename: "{{ playbook_dir }}/rhel-9-openstack.qcow2"
```

Comment:
It looks like this file is not exists and the the image upload has already been done before tests.